### ISSUE TITLE
Unable to delete tenant with shared FQDN node

### ISSUE THREAD

**Issue by dstokesf5** [F5Networks/f5-appsvcs-extension#416]
### Environment
 * Application Services Version: 3.26.0
 * BIG-IP Version: 16.0

### Summary
An FQDN node shared between two tenants using the shareNodes property prevents deleting of the tenants.

### Steps To Reproduce
1. Ensure there are no existing FQDN nodes that AS3 might re-use!
1. Submit the following declaration:
```json
{ 
    "class":"ADC",
    "schemaVersion":"3.25.0",
    "controls": {
        "class": "Controls",
        "traceResponse": true
    },
    "tenantOne": {
        "class":"Tenant",
        "appOne": {
            "class": "Application",
            "pool": {
                "class": "Pool",
                "members": [
                    {
                        "servicePort": 80,
                        "addressDiscovery": "fqdn",
                        "autoPopulate": true,
                        "hostname": "google.com",
                        "queryInterval": 0,
                        "shareNodes": true,
                        "fqdnPrefix": "node-"
                    }
                ]
            }
        }
    },
    "tenantTwo": {
        "class":"Tenant",
        "appOne": {
            "class": "Application",
            "pool": {
                "class": "Pool",
                "members": [
                    {
                        "servicePort": 80,
                        "addressDiscovery": "fqdn",
                        "autoPopulate": true,
                        "hostname": "google.com",
                        "queryInterval": 0,
                        "shareNodes": true,
                        "fqdnPrefix": "node-"
                    }
                ]
            }
        }
    }
}
```
3. Delete tenantTwo by submitting an empty tenant:
```json
{ 
    "class":"ADC",
    "schemaVersion":"3.25.0",
    "controls": {
        "class": "Controls",
        "traceResponse": true
    },
    "tenantTwo": {
        "class":"Tenant"
    }
}
```
4. Observe the following error response:
```json
        {
            "code": 422,
            "message": "declaration failed",
            "response": "01070110:3: Node address '/Common/_auto_142.250.69.206' is referenced by a member of pool '/tenantOne/appOne/pool'.",
            "host": "localhost",
            "tenant": "tenantTwo",
            "runTime": 3445
        }
```

### Expected Behavior
The DELETE request should succeed and remove tenantTwo without affecting tenantOne

### Actual Behavior
See error message in the steps to reproduce. It looks like /Common/node-google.com is appearing in the current config, but not in the desired config.

**Comment by dstokesf5** [comment]
It looks like I was only able to reproduce this once. Reviewing the error message, I see the node is in use by a tenant that is not part of the declaration in the steps to reproduce section. I am trying to see if I can find additional steps to reproduce this.

My hunch is that there is a problem with the ref-counting used on shared nodes and AS3 thinks it needs to clean up a shared node before it is supposed to.

**Comment by dstokesf5** [comment]
Updated steps to reproduce, I am seeing this consistently now. Added to the internal product backlog as AUTOTOOL-2271.

**Comment by dstokesf5** [comment]
This issue was resolved in the 3.27.0 release.
